submissions/webdevtoday/task_js_oop/app.js

```diff
@@ -0,0 +1,124 @@
+// parameters of field
+const FIELD_WIDTH = 505;
+const FIELD_HEIGHT = 606;
+
+// Enemy speed points and points out of field
+const ENEMY_MIN_SPEED = 1;
+const ENEMY_MAX_SPEED = 10;
+const ENEMY_DISTANCE_OUT_OF_FIELD_TO_THE_RIGHT = 50;
+const ENEMY_DISTANCE_OUT_OF_FIELD_TO_THE_LEFT = -100;
+
+// Player shifts to begin
+const PLAYER_CENTERING_SHIFT = 50;
+const PLAYER_INITIAL_BOTTOM_PADDING = 220;
+
+// Collision Player Move
+const LIMITING_LEFT_PADDING = 100;
+const LIMITING_TOP_PADDING = 50;
+const LIMITING_RIGHT_PADDING = 200;
+const LIMITING_BOTTOM_PADDING = 250;
+
+// Collision Win point
+const WIN_POINT_Y_1 = -14;
+const WIN_POINT_Y_2 = 0;
+
+function randomInteger(min, max) {
+    // случайное число от min до (max+1)
+    let rand = min + Math.random() * (max + 1 - min);
+    return Math.floor(rand);
+}
+// Enemies our player must avoid
+var Enemy = function(x, y) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    // x: -50; y: 50
+    this.x = x;
+    this.y = y;
+    this.speed = randomInteger(ENEMY_MIN_SPEED, ENEMY_MAX_SPEED);
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+    // all computers.
+    if (this.x > FIELD_WIDTH + ENEMY_DISTANCE_OUT_OF_FIELD_TO_THE_RIGHT) {
+        this.speed = randomInteger(ENEMY_MIN_SPEED, ENEMY_MAX_SPEED);
+        this.x = ENEMY_DISTANCE_OUT_OF_FIELD_TO_THE_LEFT;
+    }
+    ctx.drawImage(Resources.get(this.sprite), (this.x += this.speed) * dt, this.y);
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+var Player = function() {
+    this.initX = FIELD_WIDTH / 2 - PLAYER_CENTERING_SHIFT;
+    this.initY = FIELD_HEIGHT - PLAYER_INITIAL_BOTTOM_PADDING;
+    this.x = this.initX;
+    this.y = this.initY;
+    this.leftRightStepSize = 100;
+    this.upDownStepSize = 80;
+    this.sprite = 'images/char-boy.png';
+};
+Player.prototype.update = function(dt) {};
+Player.prototype.render = function() {
+    this.checkCollision();
+    ctx.drawImage( Resources.get(this.sprite), this.x, this.y );
+};
+
+Player.prototype.checkCollision = function() {
+    allEnemies.forEach(enemy => {
+        const offset = 20;
```

Comment:
20 is mysterious number.

Always consider inducing specific statements into more general. The purpose of the review is not to specify every single mistake or inconsistency.

Reply:
Sorry, I came close to a topic that was very difficult for me and now I cannot quickly get used to it, as with the previous topics.

Reply:
No worries, that's why the mentors are here and you are not on your own completely.

Reply:
That's true!